tests/testnet/network.go

```diff
@@ -346,12 +346,14 @@ func (n *testNetwork) getMessageDeliveryTransactionReceipt(
 	}
 
 	if len(logs) == 0 {
-		return nil, errors.New("Failed to find ReceiveCrossChainMessage log for relayed message")
+		return nil, errors.New("failed to find ReceiveCrossChainMessage log for relayed message")
 	} else if len(logs) > 1 {
-		return nil, errors.New("Found multiple ReceiveCrossChainMessage logs for relayed message")
+		return nil, errors.New("found multiple ReceiveCrossChainMessage logs for relayed message")
 	}
 
-	return destination.RPCClient.TransactionReceipt(ctx, logs[0].TxHash)
+	// The transaction should already be mined, but WaitMined will also wait for the eth_blockNumber
+	// endpoint to reflect the block that the transaction has been included in.
```

Comment:
Outside of PR, for [checkMessageDelivered](https://github.com/ava-labs/teleporter/blob/main/tests/testnet/network.go#L284) can it be updated to take the `TeleporterMessenger` from `destination.SubnetInfo` instead of constructing new Teleporter

Reply:
Also for the local network's `RelayMessage` can we add a similar `MessageReceived` check

Reply:
Good call, I removed `checkMessageDelivered` all together since it was reduced to a single function call with that change actually.

I don't think we need a `MessageReceived` check in `RelayMessage` given that we already find the event of the message being received in the transaction receipt.